sxt/scalar25/operation/inner_product.cc

```diff
@@ -30,6 +37,25 @@
 #include "sxt/scalar25/operation/product_mapper.h"
 
 namespace sxt::s25o {
+//--------------------------------------------------------------------------------------------------
+// async_inner_product_partial
+//--------------------------------------------------------------------------------------------------
+static xena::future<s25t::element>
+async_inner_product_partial(basct::cspan<s25t::element> lhs,
+                            basct::cspan<s25t::element> rhs) noexcept {
+  auto n = lhs.size();
+  basdv::stream stream;
+  memr::async_device_resource resource{stream};
+  memmg::managed_array<s25t::element> lhs_device_data{&resource};
+  auto lhs_fut = xendv::make_active_device_viewable(lhs_device_data, lhs);
+  memmg::managed_array<s25t::element> rhs_device_data{&resource};
+  auto rhs_dev = co_await xendv::make_active_device_viewable(rhs_device_data, rhs);
+  auto lhs_dev = co_await std::move(lhs_fut);
```

Comment:
For my understanding, if the above lines were:
```
memmg::managed_array<s25t::element> lhs_device_data{&resource};
memmg::managed_array<s25t::element> rhs_device_data{&resource};
auto lhs_dev = co_await xendv::make_active_device_viewable(lhs_device_data, lhs);
auto rhs_dev = co_await xendv::make_active_device_viewable(rhs_device_data, rhs);
```
the `co_await` would return control to the caller less efficiently. The current code is more efficient because it suspends execution as both lhs and rhs are making the data on the devices viewable. Then the final `co_await std::move(lhs_fut)` is trivial. Is that the motivation?

Reply:
Yes. If you wrote
```
auto lhs_dev = co_await xendv::make_active_device_viewable(lhs_device_data, lhs);
```
The the following line wouldn't run until the lhs_copy was complete.

But
> the final co_await std::move(lhs_fut) is trivial.

might not be trivial ... maybe the rhs copy completed before the lhs copy.

Reply:
Got it. Thanks!